src/cloudkit.cs

```diff
@@ -977,17 +1278,72 @@ interface CKReference : NSSecureCoding, NSCopying {
 		[Export ("initWithRecord:action:")]
 		IntPtr Constructor (CKRecord record, CKReferenceAction action);
 
-		[Export ("referenceAction", ArgumentSemantic.UnsafeUnretained)]
+		[Export ("referenceAction", ArgumentSemantic.Assign)]
 		CKReferenceAction ReferenceAction { get; }
 
 		[Export ("recordID", ArgumentSemantic.Copy)]
 		CKRecordID RecordId { get; }
 	}
 
+	[iOS (10,0), Watch (3,0), TV (10,0), Mac (10,12, onlyOn64 : true)]
+	[BaseType (typeof(CKSubscription))]
+	interface CKQuerySubscription : NSSecureCoding, NSCopying
+	{
+		[Export ("initWithRecordType:predicate:options:")]
+		IntPtr Constructor (string recordType, NSPredicate predicate, CKQuerySubscriptionOptions querySubscriptionOptions);
+
+		[Export ("initWithRecordType:predicate:subscriptionID:options:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string recordType, NSPredicate predicate, string subscriptionID, CKQuerySubscriptionOptions querySubscriptionOptions);
+
+		[Export ("recordType")]
+		string RecordType { get; }
+
+		[Export ("predicate", ArgumentSemantic.Copy)]
+		NSPredicate Predicate { get; }
+
+		[NullAllowed, Export ("zoneID", ArgumentSemantic.Copy)]
+		CKRecordZoneID ZoneID { get; set; }
+
+		[Export ("querySubscriptionOptions", ArgumentSemantic.Assign)]
+		CKQuerySubscriptionOptions SubscriptionOptions { get; }
+	}
+	
+	[iOS (10,0), Watch (3,0), TV (10,0), Mac (10,12, onlyOn64 : true)]
+	[BaseType (typeof(CKSubscription))]
+	interface CKRecordZoneSubscription : NSSecureCoding, NSCopying
+	{
+		[Export ("initWithZoneID:")]
+		IntPtr Constructor (CKRecordZoneID zoneID);
+
+		[Export ("initWithZoneID:subscriptionID:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (CKRecordZoneID zoneID, string subscriptionID);
+
+		[Export ("zoneID", ArgumentSemantic.Copy)]
+		CKRecordZoneID ZoneID { get; }
+
+		[NullAllowed, Export ("recordType")]
+		string RecordType { get; set; }
+	}
+	
+	[iOS (10,0), Watch (3,0), TV (10,0), Mac (10,12, onlyOn64 : true)]
+	[BaseType (typeof(CKSubscription))]
```

Comment:
add [DisableDefaultCtor] since there's a designated initializer that does not accept null for all it's parameters